src/components/sbb-menu/sbb-menu.tsx

```diff
@@ -0,0 +1,297 @@
+
+  /**
+   * Opens the menu on trigger click.
+   */
+  @Method()
+  public async open(): Promise<void> {
+    this.willOpen.emit();
+    this._setMenuPosition();
+    this._dialog.showModal();
+  }
+
+  /**
+   * Closes the menu.
+   */
+  @Method()
+  @Listen('sbb-menu-action_click')
+  public async close(): Promise<void> {
+    this.willClose.emit();
+    this._isDismissing = true;
+  }
+
+  // Resets position on window resize.
+  @Listen('resize', { target: 'window', passive: true })
+  public onResizeWindowEvent(): void {
+    if (this._open) {
+      this._setMenuPosition();
+    }
+  }
+
+  // Closes the menu on "Esc" key pressed.
+  @Listen('keydown', { target: 'window' })
+  public onEscAction(event: KeyboardEvent): void {
+    if (this._open && event.key === 'Escape') {
+      event.preventDefault();
+      this.close();
+    }
+  }
+
+  // Removes trigger click listener on trigger change.
+  @Watch('trigger')
+  public removeTriggerClickListener(
+    newValue: string | HTMLElement,
+    oldValue: string | HTMLElement
+  ): void {
+    if (newValue !== oldValue) {
+      this._menuController?.abort();
+      this._configure(this.trigger);
+    }
+  }
+
+  private _dialog: HTMLDialogElement;
+  private _triggerEl: HTMLElement;
+  private _pointerDownEventTarget: string | null;
+  private _menuController: AbortController;
+
+  public componentDidLoad(): void {
+    this._dialog = this.el.shadowRoot.querySelector('dialog');
+
+    // validate trigger element and attach event listeners
+    this._configure(this.trigger);
+
+    // close menu on interactive elements click
+    this._dismissOnInteractiveElementClick();
+
+    // listen for menu animation to end
+    this._dialog.onanimationend = (event: AnimationEvent) => this._onMenuAnimationEnd(event);
+  }
+
+  public disconnectedCallback(): void {
+    this._menuController.abort();
+  }
+
+  // Check if the trigger is valid and attach click event listeners.
+  private _configure(trigger: string | HTMLElement): void {
+    if (!trigger) {
+      return;
+    }
+
+    // check whether it's a string or an HTMLElement
+    if (typeof trigger === 'string') {
+      this._triggerEl = document.getElementById(trigger);
+    } else if (trigger instanceof window.Element) {
+      this._triggerEl = trigger;
+    }
+
+    if (this._triggerEl) {
+      this._menuController = new AbortController();
+      this._triggerEl.addEventListener('click', () => this.open(), {
+        signal: this._menuController.signal,
+      });
+    }
+
+    // close menu on backdrop click
+    this._dialog.addEventListener('pointerdown', this._pointerDownListener, {
+      signal: this._menuController.signal,
+    });
+    this._dialog.addEventListener('pointerup', this._dismissOnBackdropClick, {
+      signal: this._menuController.signal,
+    });
+  }
+
+  // Close menu at any click on an interactive element inside the <sbb-menu> that bubbles to the container.
+  private _dismissOnInteractiveElementClick(): void {
```

Comment:
I wonder if it might be better to just apply a click event listener on the wrapper and check if the event originated from one of the interactive elements instead of adding an event listerener to every element?

Reply:
That's right, it's much simpler this way. Thank you! 👍